### ISSUE TITLE
Getting a lot of nan from round 1 of feature extraction

### ISSUE THREAD

**Issue by SharhadBashar** [Juliachang/SC-CAM#23]
Hi
I am getting a lot of nan arrays or arrays with very large numbers when extracting features from round 1
Any idea what is causing this?

**Comment by wasidennis** [comment]
Can you provide more details of the problem you are encountering? Thanks.

**Comment by SharhadBashar** [comment]
Heres the error i am running into
I run step 1, extract_features.py
It runs without any issues, but the file it produces: R0_feature.npy, has bugs. It is full of nan values
When i print out feature, from `feature = feature[0].cpu().detach().numpy()` most values for the images are either nan or very large numbers
When i try to run the second step: create_pseudo_labels.py, the line k_center = KMeans(n_clusters=k_cluster, random_state=0, max_iter=max_iter).fit(X) does not work as X has a lot of nan's in it.

I printed out features, and there are a lot of nan values in it. I am not sure where these are coming from

Here is how i have my data. I have a folder called dataset, with 2 folders in it: JPEGImages and SegmentationClassAug, both has 12,031 images

Other than that, im not sure what is causing these nan values

**Comment by wasidennis** [comment]
The first step is to just extract features from the model. You may try to just input one image and see whether you can get meaningful features and debug from there.

**Comment by SharhadBashar** [comment]
Could versions of libraries used cause this?

**Comment by SharhadBashar** [comment]
Tried it with one image, works fine. tried with 2 images worked fine. Tried with 150 images: first few were fine. then the numbers in the array started to get bigger and bigger. then they became inf and finally nan
Ive attached the output for you to see. it contains the names of the images and the features that were extracted. I also attached an example image that returned all nan values
![2007_002639](https://user-images.githubusercontent.com/9545714/135704717-82f86ea4-e366-435e-bf9e-2cd837da295b.png)
[150images.txt](https://github.com/Juliachang/SC-CAM/files/7270911/150images.txt)

The image is 2007_002639.png when ran with the other 149, it returns an array of nan's. but when i run it just by itself, it returns a valid features array [0.12978874 0.22629979 0.17751996 ... 0.12295595 0.1530146  0.1952582 ]
So not sure where or what is going wrong
